packages/cli/src/databases/migrations/common/1726218295879-SeparateExecutionCreationFromStart.ts

```diff
@@ -0,0 +1,44 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+export class SeparateExecutionCreationFromStart1726218295879 implements ReversibleMigration {
+	async up({
+		schemaBuilder: { addColumns, column, dropNotNull },
+		runQuery,
+		escape,
+		runInBatches,
+	}: MigrationContext) {
+		await addColumns('execution_entity', [
+			column('createdAt').notNull.timestamp().default('NOW()'),
+		]);
+
+		await dropNotNull('execution_entity', 'startedAt');
+
+		const executionEntity = escape.tableName('execution_entity');
+		const createdAt = escape.columnName('createdAt');
+		const startedAt = escape.columnName('startedAt');
+		const id = escape.columnName('id');
+
+		const BATCH_SIZE = 500;
+
+		await runInBatches(
+			`SELECT ${id} FROM ${executionEntity}`,
+			async (batch: Array<{ id: string }>) => {
+				const ids = batch.map((row) => row.id).join(',');
+
+				if (ids.length === 0) return;
+
+				await runQuery(`
+          UPDATE ${executionEntity}
+          SET ${createdAt} = ${startedAt}
+          WHERE ${id} IN (${ids});
+        `);
+			},
+			BATCH_SIZE,
+		);
```

Comment:
Why not do this in a single sql statement which would be much faster

```sql
UPDATE execution_entity
SET createdAt = startedAt
```
?

Reply:
I actually [started with the one-liner](https://github.com/n8n-io/n8n/pull/10810/commits/6a62d475c5a39309f9d802240db7019d8f17e2aa) but thought it best to batch updates since there can be lots of executions to update. Not batching is safe then?

Reply:
We block the startup process during the migrations and everything runs in a single transaction, so I don't see any benefit in batching the updates. Better to get over it as quickly and efficiently as possible. This probably takes some time on large sqlite instances anyways

Reply:
Thanks, reverted here: [5a519fc](https://github.com/n8n-io/n8n/pull/10810/commits/5a519fc7e77cc86be89439a85ff4c0abd015c80e)

Reply:
I've seen customers with 100k+ rows in the executions table. I don't know if batching will be better here or not, but just want to make sure that doing this in a single query won't create any trouble for such customers.

Reply:
From what I understand, DB is the fastest and most efficient place to run a large update operation. The exceptions being:
1. With long and large transactions the commit can actually take a very long time. Sometimes it's better to batch the updates but for that to be helpful we would have to split them into separate transactions
2. You don't want to load the DB too much for a long period of time. Here we don't run anything else while running the migrations, so we don't need to concern with that.

Would be nice to still test this with a large DB to check how long this takes to complete